### PR TITLE
Add user function

### PR DESCRIPTION
**Pull request by elmiko**
this pull request brings in several changes that allow for a user to specify an external function which is loaded dynamically at application start up. it does change the application to be python 3 specific as the module loading functions that are being used don't seem to work the same way in python 2 and are causing serialization issues with spark.

### CODE REVIEW

**Comment by elmiko** [comment]
@ruivieira ptal

File: examples/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import json
+
+def user_defined_function(value):
+    data = value
```
Comment (ruivieira) [inline examples/wrapper.py]: perhaps adding a docstring to the example function, reminding of the API?

**Review by ruivieira — COMMENTED** [review]
@elmiko I've added a single suggestion, but lGtm!

**Comment by elmiko** [comment]
hey @ruivieira , i added the requested change. assuming this still lGty i'm gonna merge it this weekend.

**Comment by ruivieira** [comment]
@elmiko lGtm!

**Pull request merged**